lightning/src/ln/channelmanager.rs

```diff
@@ -1321,7 +1322,7 @@ pub(super) struct PeerState<SP: Deref> where SP::Target: SignerProvider {
 	/// where we complete one [`ChannelMonitorUpdate`] (but there are more pending as background
 	/// events) but we conclude all pending [`ChannelMonitorUpdate`]s have completed and its safe
 	/// to run post-completion actions.
-	in_flight_monitor_updates: BTreeMap<OutPoint, Vec<ChannelMonitorUpdate>>,
+	in_flight_monitor_updates: BTreeMap<(OutPoint, ChannelId), Vec<ChannelMonitorUpdate>>,
```

Comment:
Wondering if it would simplify anything if either the `ChannelId` or `OutPoint`? were tupled with the value instead of the key.